src/common/transformations/src/transformations/op_conversions/convert_gather_downgrade.cpp

```diff
@@ -48,6 +48,40 @@ pass::ConvertGather8ToGather7::ConvertGather8ToGather7() {
         if (!gather_v8_node)
             return false;
 
+        auto data = gather_v8_node->input_value(0);
+        auto indices_constant =
+            std::dynamic_pointer_cast<ov::op::v0::Constant>(gather_v8_node->input_value(1).get_node_shared_ptr());
+        auto axis_constant =
+            std::dynamic_pointer_cast<ov::op::v0::Constant>(gather_v8_node->input_value(2).get_node_shared_ptr());
+        if (!indices_constant || !axis_constant)
+            return false;
+
+        auto axis = axis_constant->cast_vector<int64_t>();
+        if (axis.size() != 1) {
+            return false;
+        }
+        auto axis_value = axis[0];
+        // normalize `axis` value if it is negative
+        if (axis_value < 0) {
+            axis_value = axis_value + data.get_partial_shape().rank().get_length();
```

Comment:
please check that "data.get_partial_shape().rank()" is static

Reply:
added check, thanks